diff --git a/config.tests/qpa/wayland/wayland.cpp b/config.tests/qpa/wayland/wayland.cpp
index 40ff2cb..0734bfb 100644
--- a/config.tests/qpa/wayland/wayland.cpp
+++ b/config.tests/qpa/wayland/wayland.cpp
@@ -1,5 +1,6 @@
+#include <cstddef>
 #include "wayland-client.h"
-#include "wayland-glib.h"
+//#include "wayland-glib.h"
 
 int main(int, char **)
 {
diff --git a/config.tests/qpa/wayland/wayland.pro b/config.tests/qpa/wayland/wayland.pro
index 83f5efe..266e4d6 100644
--- a/config.tests/qpa/wayland/wayland.pro
+++ b/config.tests/qpa/wayland/wayland.pro
@@ -1,4 +1,4 @@
-SOURCES = client.cpp
+SOURCES = wayland.cpp
 CONFIG -= qt
 
 QMAKE_CXXFLAGS += $$QT_CFLAGS_WAYLAND
diff --git a/src/plugins/platforms/wayland/wayland.pro b/src/plugins/platforms/wayland/wayland.pro
index ec3f9d3..28be00d 100644
--- a/src/plugins/platforms/wayland/wayland.pro
+++ b/src/plugins/platforms/wayland/wayland.pro
@@ -15,6 +15,7 @@ contains(QT_CONFIG, opengl) {
     QT += opengl
 }
 LIBS += -lwayland-client -ldrm -lxkbcommon -lEGL -lGLESv2
+INCLUDEPATH += /usr/include/libdrm
 
 include (../fontdatabases/fontconfig/fontconfig.pri)
 
