diff --git a/HAL/rtl8192/Makefile b/HAL/rtl8192/Makefile
index 34160ee..73d493b 100755
--- a/HAL/rtl8192/Makefile
+++ b/HAL/rtl8192/Makefile
@@ -65,10 +65,10 @@ EXTRA_CFLAGS += -DEEPROM_OLD_FORMAT_SUPPORT=0 		\
 		-DUSE_FW_SOURCE_IMG_FILE 		\
 		-DENABLE_DOT11D 			\
 		-DCONFIG_ASPM_OR_D3 			\
-		#					\
-		-DCONFIG_CRDA	 			\
 		-DCONFIG_CFG_80211 			\
 		-DCONFIG_RTL_RFKILL	 		\
+		#					\
+		-DCONFIG_CRDA	 			\
 		-DCONFIG_FW_SETCHAN			\
 		-DRTL8192S_WAPI_SUPPORT			\
 		-D_ENABLE_SW_BEACON 			\
