diff --git a/Makefile b/Makefile
index 2ea2d3a..1969ee9 100644
--- a/Makefile
+++ b/Makefile
@@ -39,11 +39,11 @@ subdirs-all subdirs-clean :
 	for f in $(subdirs); do $(MAKE) -C $$f $(@:subdirs-%=%); done
 
 install : $(libs) compositor
-	install -d $(libdir) $(libdir)/pkgconfig ${udev_rules_dir}
-	install $(libs) $(libdir)
-	install wayland-server.pc wayland.pc $(libdir)/pkgconfig
-	install wayland-util.h wayland-client.h $(includedir)
-	install 70-wayland.rules ${udev_rules_dir}
+	install -d $(libdir) $(DESTDIR)/$(libdir)/pkgconfig $(DESTDIR)/$(includedir) $(DESTDIR)/$(udev_rules_dir)
+	install $(libs) $(DESTDIR)/$(libdir)
+	install wayland-server.pc wayland.pc $(DESTDIR)/$(libdir)/pkgconfig
+	install wayland-util.h wayland-client.h $(DESTDIR)/$(includedir)
+	install 70-wayland.rules $(DESTDIR)/$(udev_rules_dir)
 
 clean : subdirs-clean
 	rm -f compositor *.o *.so
