diff --git a/clients/Makefile.am b/clients/Makefile.am
index d1e7134..c3bfdee 100644
--- a/clients/Makefile.am
+++ b/clients/Makefile.am
@@ -1,7 +1,7 @@
 noinst_PROGRAMS = gears flower screenshot terminal image view dnd smoke
 noinst_LTLIBRARIES = libtoytoolkit.la
 
-INCLUDES = -I$(top_srcdir)/wayland $(CLIENT_CFLAGS) $(POPPLER_CFLAGS)
+INCLUDES = -I$(top_srcdir)/wayland -I$(top_builddir)/wayland $(CLIENT_CFLAGS) $(POPPLER_CFLAGS)
 
 LDADD = libtoytoolkit.la				\
 	$(top_builddir)/wayland/libwayland-client.la	\
diff --git a/compositor/Makefile.am b/compositor/Makefile.am
index a7d121b..5933fa6 100644
--- a/compositor/Makefile.am
+++ b/compositor/Makefile.am
@@ -1,6 +1,6 @@
 noinst_PROGRAMS = compositor
 
-INCLUDES = -I$(top_srcdir)/wayland $(COMPOSITOR_CFLAGS)
+INCLUDES = -I$(top_srcdir)/wayland -I$(top_builddir)/wayland $(COMPOSITOR_CFLAGS)
 AM_CPPFLAGS = -DDATADIR='"$(datadir)"'
 
 compositor_LDADD =					\
